File: 290 Word Pattern.java
public class Solution {
    public boolean wordPattern(String pattern, String str) {
        String[] strs = str.split(" ");
        if (strs.length != pattern.length()) {
            return false;
        }
        HashMap<Character, String> hashMap = new HashMap<Character, String>();
        for (int i = 0; i < pattern.length(); ++i) {
            if (hashMap.containsKey(pattern.charAt(i))) {
                if (!hashMap.get(pattern.charAt(i)).equals(strs[i])) {
                    return false;
                }
            } else {
                if (hashMap.containsValue(strs[i])) {
                    return false;
                }
                hashMap.put(pattern.charAt(i), strs[i]);
            }
        }
        return true;
    }
}

File: 290 Word Pattern.py
class Solution(object):
    def wordPattern(self, pattern, str):
        """
        :type pattern: str
        :type str: str
        :rtype: bool
        """
        strs = str.split(' ')
        if len(pattern) != len(strs):
            return False
        for i in xrange(len(pattern)):
            for j in xrange(len(pattern)):
                if pattern[i] == pattern[j] and strs[i] != strs[j]:
                    return False
                if pattern[i] != pattern[j] and strs[i] == strs[j]:
                    return False
        return True
